Array1Product.java

```diff
@@ -0,0 +1,26 @@
+// Time Complexity : O(n)
+// Space Complexity : O(1)
+// Did this code successfully run on Leetcode :
+// Any problem you faced while coding this :
+
+
+// Your code here along with comments explaining your approach
+
+class Solution{
+	public int[] Array(int[] nums ){
+		int[] result=new int[];
+		int n=nums.length;
+		int rp=1;
+		result[0]=1;
+		for(int i=1;i<=n;i++){
+			rp=rp*nums[i];
+			result[i]=rp;
+		}
+		rp=1;
+		for(int j=2;j>n;j--){ //why is it n-2?
```

Comment:
Because you are multiplying with its next element, if you try to get element next element for index(n-1) it will give out of bound index exception as array indices are from 0 to n-1 only so.

Reply:
Please re-write again the second one because this will give you wrong answer.

Reply:
Did you try to run the code on Leetcode?